### DISCUSSION TITLE
Team Agreement

### DISCUSSION THREAD

**Discussion by metazool** [NERC-CEH/rse_group#23] [General]
When we met in-person as a group in Lancaster one item on the agenda was a Team Agreement - see #1 

[What is a Team Agreement + the 9 Steps to Create One](https://www.runn.io/blog/team-agreement) - picked the first search result from the list as an overview, I haven't fully read this. At my previous gig we had a "Team Charter" and it was a successful document for negotiating with other groups who came to us with requests for consultancy and collaboration - _this is who we are, and why we work on what we do, and how we aim to work on it_.

 What are our options for taking it from an abstract draft to an agreement that can be collectively got behind? Can it work asynchronously or would a short "workshop" session be a lot more efficient?

**Comment by matthewcoole** [comment]
This was an interesting read. I've come up with some random thoughts - not completely invested in all these ideas, but maybe we can start to get the ball rolling.

## Team Aims/Goals
- What are our objectives?
- Contribute to the design and implementation of software solutions that help to support quality, open science across UKCEH?
- Help develop the digital skills of those in UKCEH. Starting from the inside out.
	- Develop and share skills within the RSE team.
	- "" within the EDS team.
	- "" within the pollution science area.
	- "" within UKCEH.
	- "" with partner organisations.
## Evolution
- We will not bind ourselves to one way of working, one set of tools, one line of thought forever - the opportunity to change and evolve will be baked in.
- Anyone who is unhappy with how anything within the team is working should be able to speak up without fear.
- Seniors, who are willing, will act on behalf of anyone (including other seniors) to bring forth concerns that people may prefer to discuss privately and make sure that they are considered by the group (confidentially if needed).

## Working Practices
### General
- We will align ourselves where possible with the current operating standards of UKCEH, this may include but is not limited to:
	- Using agile process when working with the development team.
	- Potentially adopting a similar sprint cycle for other project work (possibly 4 week cycles?)
	- Adopting a show and tell practice to demo our work to each other - possibly once every 4 weeks as well?
### Communication
- We will always be respectful to each other in all in-person, written, and hybrid communication.
- Asynchronous communication will always be preferred when it is feasible.
- We will foster a safe environment - people will not be chastised for not knowing something or for making honest mistakes.
#### Meetings
- Meetings will be minimized where possible, in both length and frequency.
- Meetings will have a clear focus and agenda and all will work to keep us all on topic during meetings.
- Meetings will, wherever possible be scheduled in core working hours i.e. 10am-12noon, 2pm-4pm, Mon-Fri.
- Team members will "be present", "show personal interest", "recognize teammates".

**Reply by metazool** [reply]
I really like these contributions @matthewcoole - the spirit of them reminds me of the Recurse Center material on self-directives and social rules: https://www.recurse.com/manual#sec-environment 

We could add a reference to the Turing Way principles of reproducible (data) science https://www.turing.ac.uk/research/research-projects/turing-way to get a little more concrete about the norms of delivered work we aim to reach (tests, CI, documentation etc)

The one word i'd change is "We will always be respectful" to something more like "We will always be considerate" (or even "kind", though that's not a word one hears often enough in a professional setting! 

The reason for this is the semantic ambiguity to _respect_, related to cultural status. It's been really well put in a quote I've seen fly by on the Fediverse a few times but can't easily turn up in a search. The quote goes something like "people use _respect_ in more than one way - respect for a person, and respect for an authority - and sometimes it's used to imply, _if you don't respect me as an authority, i won't respect you as a person_." The sentiment was expressed more pithily than that, though!

**Reply by matthewcoole** [reply]
I like "always be kind" as a good team ethos.

I also really like the idea of linking things to the Turing Way. Perhaps we could have a stripped down set of things described in the Turing Way to allow us a little flexibility depending on the nature of what we're working on (and the practicalities of what we can deliver for different projects).

**Reply by metazool** [reply]
If this is an opportunity to load up the document with what's ideal as well as what's non-arguable, then I'd like to add 

* Open by default - we aim to hold technical discussions in open forums, and work on software projects which are open licensed and publically available

**Reply by jmarshrossney** [reply]
I strongly agree with everything after the dash! As we've discussed before and surely will continue to do so, I do feel that 'open by default' needs a bit of unpacking though.

**Reply by longr** [reply]
I feel like the Turing Way should not be linked in with this.   The team agreement is how we operate.  There is space for a Turing way like document to be co-designed that sets out a wider how research software should be created within CEH, which the team agreement will then say we will adhere to this in any software we have complete control over, and advocate that those within CEH should adhere to this if they want us to work them (or is that too harsh)

**Comment by mattjbr123** [comment]
Ok, let's have a go at "disneyfying" this and putting down some thoughts, however unrealistic they may be.
Mostly, I really really like what you've put down @matthewcoole in particular the thought that's gone in to the "how we treat each other" bits. 

- Loving the "default to open" idea that's come out of some of discussions here and elsewhere, woud love that to be in the agreement somewhere
- Finding the right words for the "be kind to each other, we all make mistakes and are learning here and are busy people and forget things" bit might be tricky, but deserves some time and finessing as I think this is the most important thing of all in the agreement, it underpins everything in a team. Should definitely add inclusion/inclusivity into this bit. 
- Flexibility. "This is how we do things now but we reserve the right to change this as our thinking/teamwork/the industry best practices change". Probably needs slightly less aggressive wording. 
- We aim to make our work as open and reproducible as possible, with some examples to demonstrate what this might look like in practice
- In fact, having as many examples as possible is a good thing. These sorts of documents can get very abstract very quickly imo, so some examples of a "this is what we mean when we say that" always helps me actually engage with the document

My mind has gone blank for now, but I imagine things will occur to me later...


Side note, I remember hearing word of there being a team charter that at least one of the groups in EDS mentioned having, would be good to have a look at it but no idea where to find it just yet

**Reply by mattjbr123** [reply]
Oh I thought of another thing (knew that would happen!), definitely 'disney': having just one place to monitor for communications, and another for task tracking if necessarily separate.

**Reply by jmarshrossney** [reply]
> Flexibility. "This is how we do things now but we reserve the right to change this as our thinking/teamwork/the industry best practices change". Probably needs slightly less aggressive wording

I definitely agree with this! Something something "principles rather than specific tools" being important. But maybe leave open the idea that some principles might change in future..?

> having just one place to monitor for communications, and another for task tracking if necessarily separate.

Matt talked about minimising meetings, we might also talk about minimising unnecessary administrative burden, but what about something like "we recognise the reality of digital fatigue and oppressive modes of digital communication (owtte) and therefore aim to keep communications clear, purposeful, and via as few different channels as is feasible" or something

**Reply by mattjbr123** [reply]
Love the digital fatigue idea and wording! Definitely something I struggle with...

**Comment by albags** [comment]
I like a lot what the rest of the team has written, especially about openness and communication. Here are some of my thoughts/wishes.

- Having a clear mission like: "Our mission is to develop open, reproducible, robust, innovative software solutions that advance scientific research in ecology and hydrology and contribute to the academic community."

- Communication and meetings: follow agile principles, to some extent. Because we won't be working all on the same projects. Totally agree with @matthewcoole on how we should treat each other. 

- Collaboration and coding: I really would like to learn more about mob programming and pair programming, @metazool has already started putting some documentation. Aim to follow best practices and FAIR principles. 

- Professional development: it would be good to attend at least one professional development workshop or conference per year. And we all should have opportunities to do training. Not only receive the training but also be a reference for the rest of CEH about software development. 

- Review and adapt: I like the concept of flexibility and maybe try some things and keep revising if they are working for us. 

I will keep thinking about other points.

**Reply by mattjbr123** [reply]
I like the "be a reference for ... CEH about software development" and the training/professional development requirement that goes with it

**Reply by jmarshrossney** [reply]
> Having a clear mission like: "Our mission is to develop open, reproducible, robust, innovative software solutions that advance scientific research in ecology and hydrology and contribute to the academic community."

Can we have multiple (though not too many) missions? It's true that our roles in supporting learning and skills development among colleagues is kind of a sub-goal of the one you wrote above, but it might be more practically useful to treat it as a different mission.

E.g. if we are evaluating ourselves in one year's time we might want to say "how are we doing in relation to Mission 1: deveop ... software? How about Mission 2: support learning...? Mission 3: make distinctive contributions to the scientific literature?"

**Comment by jmarshrossney** [comment]
## Team agreement 'Disney' version

I like many things that have already been posted, so rather than post 'my version' I'll just focus on the few things I would dream of adding or doing differently. Mainly absent are things to do with working practices which I think have been well-covered elsewhere.

This is just a collection of half-finished and probably over-elaborate ideas. Treat as a pick and mix!

### Learning, teaching and knowledge

I would like the team agreement to include something about the way we understand ourselves as teachers and learners, both individually and in the context of the organisation.

On an individual level...

- We see ourselves as 'lifelong learners' and commit to learning in such a way that expands or challenges our present ways of thinking.
- We view this continual as a core part of our work and essential to the success of the RSE initiative, and therefore that it can and should be something that occurs during working hours with realistic time allocated to it.

For teaching, ideally I'd like to avoid language that describes us as 'knowledge-holders' and describes what we do as a transferral of our knowledge to our colleagues. I would prefer language that emphasises some or all of the following: (a) that we are both teachers and learners simultaneously, (b) that the learning we are interested in is social in nature, (c) connected to the previous two point, that we are also trying to learn from those who we are teaching.

My 'Disney dreamer' version looks something like this:

- We recognise that not all of the learning needs of the organisation can be adequately met by structured learning activities such as tutorials and workshops. There is a need for ... ==TODO==
- Our ambition is therefore to create, foster, and take leadership roles in a community of practice centred on research software at UKCEH, which can support emergent social structures that in turn support social learning processes
- Specifically, our roles include
	- Understanding our community at both individual and collective levels - where are its strengths, and where are there gaps?
	- Ensuring that the learning needs of the community are met in an equitable manner
	- Steering the community towards alignment of our local activities with broader organisational ambitions
	- Tracking important developments from the wider world, and synthesising them for the benefit of the community 
	- Fostering inter- and intra-community connections
	- Organising structured activities for community-building and learning
	- Creating tangible learning resources based on the specific needs of the community
	- Promoting community-oriented learning practices, and encouraging members to 'learn in the open' by e.g. reaching out to other members with similar issues/needs, documenting their problems and solutions

### Acknowledge what is unique about UKCEH

What makes us different from an RSE group working for the research wing of a private company? Or for a University?

(a) the science content:

- We are committed to learning about the scientific activities we are contributing to, and recognise that this is an important part of our job

(b) the funding:

- We understand being (largely) publicly funded means we have certain social responsibilities beyond those that apply to employees of private companies. This includes... ==TODO==

(c) the reputation of UKCEH for rigorous, transparent and ethical scientific research

- We will not act in a way that undermines the reputation of UKCEH for scientific research (ok but whistleblowing...)
- We will carefully consider who we collaborate with and whose services we employ, referring for guidance to... ==TODO==

(d) the values and ambitions of UKCEH:

- We will act in a way that is congruent with the values and ambitions of UKCEH, i.e. for the benefit of 'the people' (who?) and 'the environment' (which parts?) (basically not for profit, individual advancement etc.)

I kind of want this to be important, but haven't quite got a handle on the specifics yet...

### Workloads

What is an acceptable workload? What is a reasonable expectation for working outside of our contracted hours? How will we support each other to manage workloads and ensure that they stay within the acceptable bounds?

I think we have a right not to work substantially more than our contracted hours, taken over a long period of time (months?). At the same time I think it's reasonable that we do not always clock off at a fixed time each day irrespective of work unless there are reasonable grounds to do so, which could include e.g. pick-up from school, pre-arranged trip, or that we have done too much overtime already and there has not been a mutually agreed way to bring the workload down.

Not sure if workloads is really something that should be in our team agreement, but the ways in which we will support each other does seem relevant.

- We will support each other in managing workloads and navigating complex working arrangements
- We will speak with one voice if there are unreasonable burdens being placed on any one of us 

### Work culture and ethics

We should include something about work culture and our position as leaders and potential role models.  challenging exploitative practices even when it involves going against cultural norms in research/academia.

- We recognise that we hold positions of leadership in our workplace and professional community and that our actions influence the culture of that workplace and that professional community. Therefore...
- We will not support or engage in practices that exploit others, especially more junior or precarious researchers. For example,
	- Offloading work onto others for our own convenience
	- Failing to recognise when a workload has become unreasonably heavy
	- Taking credit for work whilst simultaneously seeking to minimise our own workload.
- Not only will we decline to support such exploitative practices, but we will openly challenge them in a constructive manner
- This includes occasions where we, or UKCEH, would stand to benefit from such practices, for example when supervising students, or conversely when there are professional risks to challenging such practices

### Working in the open, but usefully

I do really like the 'working in the open by default' principle, but perhaps I would add some of these caveats. Mainly I see 'working *usefully* in the open' as an unsolved problem (at least unsolved by us) which I would like us to commit to keep thinking about.

- We intend to work 'in the open' by default, which means... ==TO DO==
- However, we recognise that there are potential risks (perceived or actual) of working in the open, particularly when the work involves or relates to colleagues outside of our group who may work under different circumstances or otherwise feel differently about working in the open (e.g. colleagues who are under pressure to publish in highly competitive fields)
- We also acknowledge that working in the open is not enough; we will strive to work in the open in a way that is also accessible, discoverable, and useful to both our future selves and the wider world
- We furthermore recognise that 'open by default' can potentially frustrate this aim if much of what is published is vanishingly unlikely to be useful because it
	- Is a transient artefact of a process that will likely never need to be retraced
	- Could be misleading if read in isolation (e.g. wrong assumption that people will read everything)
	- Is not clearly separable from more important information / decreases the signal/noise ratio in important situations
	- Presents an accessibility challenge for those who may be slower to digest significant amounts of non-curated textual information, such as people with dyslexia and non-native speakers
- Hence, we will strive to... ==TO DO==

### Wider context of automation and future 

Since much of our work will have something to do with automation and modernisation of existing practices, I feel we should say something about the things we will hold in our heads as we move forward in a world in which people increasingly perceive threats to their job security and even sense of identity.

- We recognise that our voices carry (a modest amount of) weight in wider societal discussions about automation and the future of work
- We will refrain from making or endorsing exaggerated or unevidenced claims of the effectiveness and robustness of automated systems, even (and especially) when there may be personal gains to such actions
- Rather, we will be champions of the expertise of experienced scientists and practitioners
- We intend to promote semi-automated working practices that 'keep humans in the loop' - and as far as possible the same humans who have the depth of experience needed to be effective critics of such systems
- We acknowledge our privileged position in that, for the time being at least, the skills we have are valued in a way that is reflected in the labour market, and that we need to empathise and engage with others in different positions to ... ==TODO==

**Reply by jmarshrossney** [reply]
- We will not work in silos, but strive towards modes of working that allow us to move forwards as a team with minimal friction (while still recognising that each of us are distinct and non-interchangeable)

^ this is probably important, and something I'm acutely conscious of needing to work on myself.

**Reply by longr** [reply]
I understand where you are going with Workloads, (and I may have misunderstood), but having spent many years in a research environment and seen workload creep, and the blurring of worklife balance, I think this needs to be a hard line.  There is the option of TOIL, but I strongly feel this should be driven by our needs, not the projects.  Ideally we will be on multiple project (2 at a time).   This deadline / split should not be hard and if needed we can work more on one project one week, and catch this backup later if needed.    We should keep track of this though, as if we need to constanly work more ours on a project then it maybe its RSE-hours was estimated incorrectly.

**Comment by matthewcoole** [comment]
Draft team agreement based on this discussion: https://github.com/NERC-CEH/rse_group/pull/33
